src/erlfmt_format.erl

```diff
@@ -259,9 +259,24 @@ do_expr_to_algebra({clauses, _Meta, Clauses}) ->
     clauses_to_algebra(Clauses);
 do_expr_to_algebra({body, _Meta, Exprs}) ->
     block_to_algebra(Exprs);
-do_expr_to_algebra({sigil, _Meta, Prefix, Content, Suffix}) ->
+do_expr_to_algebra({sigil, _Meta, Prefix, {string, ContentMeta, _Value}, Suffix}) ->
     PrefixDoc = concat(<<"~">>, do_expr_to_algebra(Prefix)),
-    concat(concat(PrefixDoc, do_expr_to_algebra(Content)), do_expr_to_algebra(Suffix));
+    Text = erlfmt_scan:get_anno(text, ContentMeta),
+    SigilDoc = concat(concat(PrefixDoc, string(Text)), do_expr_to_algebra(Suffix)),
+    case string:split(Text, "\n", all) of
+        % Triple-quoted sigils don't need force_breaks
+        ["\"\"\"" ++ _ | _] -> SigilDoc;
+        % Single-line sigils don't need force_breaks
+        [_] -> SigilDoc;
+        % Multi-line non-triple-quoted sigils need force_breaks
+        _ -> concat([force_breaks(), SigilDoc])
+    end;
+do_expr_to_algebra({sigil, _Meta, Prefix, {Atomic, ContentMeta, _Value}, Suffix}) when
```

Comment:
Per the parser, sigil can only have string inside. This clause will never be executed.

https://github.com/WhatsApp/erlfmt/blob/1292f7c96b0d055f2e4c1facfdb54802e2ebdfbb/src/erlfmt_parse.yrl#L653

Reply:
[08de55a](https://github.com/WhatsApp/erlfmt/pull/371/commits/08de55a5354dbddda7aeb754ac9363e7b979b0e7)